functions/src/index.ts

```diff
@@ -22,15 +24,89 @@ export const getNearbyPins = functions.https.onCall(
         "getNearbyPins must be called with a latitude and longitude."
       );
     }
+
     const pins = await getPinsNearby(latitude, longitude, NUM_MILES_NEARBY);
     return Object.fromEntries(pins.map((pin) => [pin.id, pin.data().location]));
   }
 );
 
+// TODO: add anti-spoof check before dropping pin
+export const dropPin = functions.https.onCall(
+  async ({ content, caption, type, latitude, longitude }, context) => {
```

Comment:
Since we're in typescript, are we able to make this a `Pin` type that matches our Java object?